matscipy/fracture_mechanics/crack.py

```diff
@@ -1387,20 +1859,217 @@ def frame(idx):
                 p.set_data(
                     [a.positions[region == r, 0], a.positions[region == r, 1]])
             # update bonds - requires a neighbour list recalculation
-            i, j = neighbour_list('ij', a, cutoff)
-            i, j = np.array([(I, J) for I, J in zip(i, j) if
-                             region[I] in regions and region[J] in regions]).T
-            lines = list(zip(a.positions[i, 0:2], a.positions[j, 0:2]))
-            lc.set_segments(lines)
+            if cutoff is not None:
+                i, j = neighbour_list('ij', a, cutoff)
+                i, j = np.array([(I, J) for I, J in zip(i, j) if
+                                region[I] in regions and region[J] in regions]).T
+                lines = list(zip(a.positions[i, 0:2], a.positions[j, 0:2]))
+                lc.set_segments(lines)
             # update crack tip indicator
             tip.set_data([self.cryst.cell[0, 0] / 2.0 + x[idx, -2]],
                          [self.cryst.cell[1, 1] / 2.0])
             return blob, plot_elements, lc, tip
 
         if frames is None:
             frames = range(0, len(x), 100)
+
         return FuncAnimation(fig, frame, frames)
 
+    @counted
```

Comment:
what does this  `@counted` decorator do?

Reply:
@counted just keeps track of how many times the write_atoms_to_file function is called, this allows the function to save .xyz files with consecutive names each time it is called. Admittedly, I now realise the same effect could have been achieved by having an instance attribute for sinclair_crack which keeps track of how often the function is called

Reply:
ok - not too fussed by an explicit instance attribute might be clearer